sql/importer: flake in TestCSVImportCanBeResumed [ingestKVs hangs with BulkAdderFlushesEveryBatch]

Found here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelExtendedCi/7502297?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
Failed
=== RUN   TestCSVImportCanBeResumed
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/375538c3f706ce59f4e97af17139af4f/logTestCSVImportCanBeResumed2019049709
    test_log_scope.go:79: use -show-logs to present logs inline
    import_processor_test.go:739: Resume pos: 10
    import_processor_test.go:745: not enough memory available to create a BulkAdder: root: memory budget exceeded: 33554432 bytes requested, 101984256 currently allocated, 134217728 bytes in budget
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/375538c3f706ce59f4e97af17139af4f/logTestCSVImportCanBeResumed2019049709
--- FAIL: TestCSVImportCanBeResumed (1.16s)
```

cc @rafiss for triage

Jira issue: CRDB-21446

I applied a memory limit in https://github.com/cockroachdb/cockroach/pull/92689, but after that this test fails under stress with a new error

```
--- FAIL: TestCSVImportCanBeResumed (6.96s)
    test_log_scope.go:161: test logs captured to: /tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestCSVImportCanBeResumed3015423266
    test_log_scope.go:79: use -show-logs to present logs inline
    import_processor_test.go:741: Resume pos: 10
    import_processor_test.go:755: -- test log scope end --
    import_processor_test.go:755: Leaked goroutine: goroutine 1657 [semacquire]:
        sync.runtime_Semacquire(0x11c41c5?)
                GOROOT/src/runtime/sema.go:62 +0x25
        sync.(*WaitGroup).Wait(0xc0035cddf0?)
                GOROOT/src/sync/waitgroup.go:139 +0x52
        golang.org/x/sync/errgroup.(*Group).Wait(0xc0041d34c0)
                golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:53 +0x27
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.Wait({0xc0041d34c0?, {0x6a60950?, 0xc0041d3480?}})
                github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:144 +0x4a
        github.com/cockroachdb/cockroach/pkg/sql/importer.runImport({0x6a609f8, 0xc000f14780}, 0xc001b60200, 0xc0041bdd50, 0xc004f94f60, 0xc0041d33c0?)
                github.com/cockroachdb/cockroach/pkg/sql/importer/read_import_base.go:127 +0x7c6
        github.com/cockroachdb/cockroach/pkg/sql/importer.(*readImportDataProcessor).Start.func1()
                github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor.go:183 +0x8b
        created by github.com/cockroachdb/cockroach/pkg/sql/importer.(*readImportDataProcessor).Start
                github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor.go:181 +0xeb
        Leaked goroutine: goroutine 1659 [semacquire]:
        sync.runtime_Semacquire(0x11c41c5?)
                GOROOT/src/runtime/sema.go:62 +0x25
        sync.(*WaitGroup).Wait(0xc003565808?)
                GOROOT/src/sync/waitgroup.go:139 +0x52
        golang.org/x/sync/errgroup.(*Group).Wait(0xc0041d35c0)
                golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:53 +0x27
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.Wait({0xc0041d35c0?, {0x6a60950?, 0xc0041d3580?}})
                github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:144 +0x4a
        github.com/cockroachdb/cockroach/pkg/sql/importer.ingestKvs({0x6a60950?, 0xc0041d3480?}, 0xc001b60200, 0xc0041bdd50, 0xc004f94f60, 0xc004680600)
                github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor.go:553 +0x1028
        github.com/cockroachdb/cockroach/pkg/sql/importer.runImport.func2({0x6a60950, 0xc0041d3480})
                github.com/cockroachdb/cockroach/pkg/sql/importer/read_import_base.go:108 +0x6a
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
                github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
        golang.org/x/sync/errgroup.(*Group).Go.func1()
                golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
                golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:72 +0xa5
        Leaked goroutine: goroutine 1661 [chan send]:
        github.com/cockroachdb/cockroach/pkg/sql/importer.ingestKvs.func3()
                github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor.go:468 +0x311
        github.com/cockroachdb/cockroach/pkg/sql/importer.ingestKvs.func5({0x6a60950, 0xc0041d3580})
                github.com/cockroachdb/cockroach/pkg/sql/importer/import_processor.go:547 +0x2cf
        github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
                github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
        golang.org/x/sync/errgroup.(*Group).Go.func1()
                golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
```

The `ingestKvs` function is getting stuck in the `BulkAdderFlushesEveryBatch` testing knob: https://github.com/cockroachdb/cockroach/blob/a16106e3c0098a8e02ad62d9b5b6f2097bae7718/pkg/sql/importer/import_processor.go#L544-L548

@adityamaru or @dt - do you have any idea why this might happen?

--- a/pkg/sql/importer/import_processor_test.go
+++ b/pkg/sql/importer/import_processor_test.go
@@ -42,7 +42,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/testutils/datapathutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
@@ -649,9 +648,6 @@ func TestCSVImportCanBeResumed(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	// Flaky test.
-	skip.WithIssue(t, 91828)
-
 	defer setImportReaderParallelism(1)()
 	const batchSize = 5
 	defer TestingSetParallelImporterReaderBatchSize(batchSize)()
@@ -674,6 +670,7 @@ func TestCSVImportCanBeResumed(t *testing.T) {
 	defer s.Stopper().Stop(ctx)
 
 	sqlDB := sqlutils.MakeSQLRunner(db)
+	setSmallIngestBufferSizes(t, sqlDB)
 	sqlDB.Exec(t, `CREATE DATABASE d`)
 	sqlDB.Exec(t, "CREATE TABLE t (id INT, data STRING)")
 	defer sqlDB.Exec(t, `DROP TABLE t`)

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -2944,8 +2944,8 @@ func TestImportRetriesBreakerOpenFailure(t *testing.T) {
 	}
 
 	sqlDB := sqlutils.MakeSQLRunner(tc.ServerConn(0))
+	setSmallIngestBufferSizes(t, sqlDB)
 	sqlDB.Exec(t, `CREATE TABLE t (a INT, b STRING)`)
-	sqlDB.Exec(t, `SET CLUSTER SETTING kv.bulk_ingest.pk_buffer_size = '16MiB'`)
 	var tableID int64
 	sqlDB.QueryRow(t, `SELECT id FROM system.namespace WHERE name = 't'`).Scan(&tableID)
 
@@ -5271,6 +5271,15 @@ func TestImportControlJobRBAC(t *testing.T) {
 	}
 }
 
+// setSmallIngestBufferSizes lowers the initial buffering adder ingest
+// size to allow import jobs to run without exceeding the test memory
+// monitor.
+func setSmallIngestBufferSizes(t *testing.T, sqlDB *sqlutils.SQLRunner) {
+	t.Helper()
+	sqlDB.Exec(t, `SET CLUSTER SETTING kv.bulk_ingest.pk_buffer_size = '16MiB'`)
+	sqlDB.Exec(t, `SET CLUSTER SETTING kv.bulk_ingest.index_buffer_size = '16MiB'`)
+}
+
 // TestImportWorkerFailure tests that IMPORT retries after the failure of a
 // worker node.
 func TestImportWorkerFailure(t *testing.T) {
@@ -5289,11 +5298,7 @@ func TestImportWorkerFailure(t *testing.T) {
 	defer tc.Stopper().Stop(ctx)
 	conn := tc.ServerConn(0)
 	sqlDB := sqlutils.MakeSQLRunner(conn)
-
-	// Lower the initial buffering adder ingest size to allow import jobs to run
-	// without exceeding the test memory monitor.
-	sqlDB.Exec(t, `SET CLUSTER SETTING kv.bulk_ingest.pk_buffer_size = '16MiB'`)
-	sqlDB.Exec(t, `SET CLUSTER SETTING kv.bulk_ingest.index_buffer_size = '16MiB'`)
+	setSmallIngestBufferSizes(t, sqlDB)
 
 	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if r.Method == "GET" {

